pkg/internal/controller/controller.go

```diff
@@ -83,8 +85,12 @@ type Controller struct {
 	// startWatches maintains a list of sources, handlers, and predicates to start when the controller is started.
 	startWatches []watchDescription
 
-	// Log is used to log messages to users during reconciliation, or for example when a watch is started.
-	Log logr.Logger
+	// LogConstructor is used to construct a logger to then log messages to users during reconciliation,
+	// or for example when a watch is started.
+	LogConstructor func(request *reconcile.Request) logr.Logger
```

Comment:
Please document that it needs to handle being called with a nil `request`

Reply:
Done